Signal/src/ViewControllers/HomeView/HomeViewController.m

```diff
@@ -47,7 +47,7 @@ @interface HomeViewController () <UITableViewDelegate, UITableViewDataSource, UI
 @property (nonatomic) UISegmentedControl *segmentedControl;
 @property (nonatomic) id previewingContext;
 @property (nonatomic) NSSet<NSString *> *blockedPhoneNumberSet;
-
+@property (nonatomic, readonly) NSCache<NSString *, ThreadModel *> *threadModelCache;
```

Comment:
`ThreadModel` seems pretty cheap to instantiate, but I wonder if we should keep around the complete list of `ThreadModel` instances for the current load window, just like conversation view.

Reply:
That would definitely be a nice future enhancement! 

Even with all the threadModel's cached, a significant portion of our remaining MainThread time is fetching threads from the Mapping, even though we only use them to get the cache key for the threadModel.